traitsui/editors/shell_editor.py

```diff
@@ -50,16 +50,21 @@ def init(self, parent):
         value = self.value
         if self.factory.share and isinstance(value, dict):
             locals = value
-        self._shell = shell = PythonShell(parent, locals=locals)
+        self._shell = shell = PythonShell(parent)
         self.control = shell.control
+        shell.interpreter().locals = locals
         if locals is None:
             object = self.object
             shell.bind("self", object)
             shell.observe(
                 self.update_object, "command_executed", dispatch="ui"
             )
             if not isinstance(value, dict):
-                object.observe(self.update_any, dispatch="ui")
+                object.observe(
+                    self.update_any,
+                    match(lambda name, ctrait: True),
```

Comment:
i'm not sure how this is working because we're not importing `match` from traits observe here. maybe it got imported by some other module before we get here.